src/Magnum/Trade/MeshData.h

```diff
@@ -147,6 +147,23 @@ enum class MeshAttribute: UnsignedShort {
      */
     ObjectId,
 
+    /**
+     * Weights. Type is usually @ref VertexFormat::Vector4, but can be also
+     * @ref VertexFormat::Vector4h, @ref VertexFormat::Vector4ubNormalized
+     * or @ref VertexFormat::Vector4usNormalized.
+     * Corresponds to @ref Shaders::Generic::Weights.
+     * @see @ref MeshData::weightsAsArray()
+     */
+    Weights,
+
+    /**
+     * Joint IDs. Type is usually @ref VertexFormat::Vector4ui, but can be also
+     * @ref VertexFormat::Vector4us or @ref VertexFormat::Vector4ub.
+     * Corresponds to @ref Shaders::Generic::JointIds.
+     * @see @ref MeshData::jointIdsAsArray()
+     */
+    JointIds,
```

Comment:
I'm wondering -- would it make sense to support also one/two/three-component weights and joints? I guess it would be useful for file size savings. Can you check if there are example glTF files with less than 4 weights?

This won't complicate the implementation too much, a similar thing is already done for colors for example.

Reply:
The accessor is always a 4-component vector, so in case there is a file like this, it would have 4 but the unused components would be zero weight

Reply:
In (unextended) glTF yes, but in general not. I don't see why glTF couldn't support `SCALAR`, `VEC2` and `VEC3` also.

Reply:
It would be just JointIds always, but allowing also one/two/three-component types instead of just Vectot4ui.

`MeshData::jointIdsInto()` will then need to zero-fill the unused components like `colorsInto()` does for three-component colors or `positions3DInto()` does for two-component positions, yes.